iceoryx_posh/include/iceoryx_posh/internal/popo/wait_set.inl

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_POSH_POPO_WAIT_SET_INL
+#define IOX_POSH_POPO_WAIT_SET_INL
+
+namespace iox
+{
+namespace popo
+{
+template <typename T>
+inline cxx::expected<TriggerHandle, WaitSetError>
+WaitSet::acquireTrigger(T* const origin,
+                        const cxx::ConstMethodCallback<bool>& triggerCallback,
+                        const cxx::MethodCallback<void, uint64_t>& invalidationCallback,
+                        const uint64_t triggerId,
+                        const Trigger::Callback<T> callback) noexcept
+{
+    static_assert(!std::is_copy_constructible<T>::value && !std::is_copy_assignable<T>::value
+                      && !std::is_move_assignable<T>::value && !std::is_move_constructible<T>::value,
+                  "At the moment only non copyable and non movable origin types are supported! To implement this we "
```

Comment:
After the copy operation one class would be attached to a waitset and another class would not be attached. 

- It would be possible to save the waitset ptr in some member and reattach the second copy again to the waitset but what should we do if the waitset is already full? ErrorHandler? Terminate? It is the copy assignment/ctor so returning an expected is not an option.

- Move can be handled a little bit easier since it is possible to update the origin of a trigger. But I would see this as a feature for another pull request, but I will add this to the issue.